Fix and clean lametric

MartinHjelmare opened the pull request:
## Description:
* Add missing DEPENDENCIES in notify platform.
* Don't overwrite notify DOMAIN.
* Remove not needed method in component manager class.
* Return consistently depending on found devices in setup of component.

**Related issue (if applicable):**
fixes #8829
closes #10230 
fixes #9420 

## Checklist:
If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

MartinHjelmare committed:
Fix and clean lametric

* Add missing DEPENDENCIES in notify platform.
* Remove not needed method in component manager class.
* Don't overwrite notify DOMAIN.
* Return consistently depending on found devices in setup of component.

MartinHjelmare commented:
@frittes if your're able to test this PR that would be great. I can't test it myself.

frittes commented:
@MartinHjelmare I am testing and it works. Thanks. What do you think, should it also work after oauth token expires?

MartinHjelmare commented:
Thanks for testing! I could add some logic that uses the reconnect method available. I'll fix that later tonight.

MartinHjelmare committed:
Get new token if token expired

MartinHjelmare commented:
I've added logic to call the `get_token` method of the manager if there's a `TokenExpiredError`. I hope this fetches a new token, but I'm not experienced in oauth2, so I'm not sure if that's enough or if a new session is required. The other option is to create a new manager instance, which creates a new session from scratch, using the `reconnect` method, but I'd like to avoid that if possible.

Please test with debug level logging for `homeassistant.components.notify.lametric`. I've added debug log "Token expired, fetching new token" before the call to get token. Of course, if it will fails, you'll get an error after that, so at least it should be obvious if it doesn't work. But I'm not sure how long until the token expires.

MartinHjelmare committed:
Add debug log for getting new token

MartinHjelmare committed:
Clean up

frittes commented:
@MartinHjelmare I am testing and give you a response tomorrow. Don´t know when the token expires...

frittes commented:
@MartinHjelmare After one day runtime of ha, it still works! 

Is it possible to make the notification sound working? No problem to test...

MartinHjelmare commented:
Great! Have you seen any debug log lines from the  platform about fetching new token?

Sorry, I don't know anything about notification sound. What do you mean?

frittes commented:
@MartinHjelmare Yes, log shows

`2017-11-13 17:43:44 DEBUG (Thread-6) [homeassistant.components.notify.lametric] Token expired, fetching new token`

About the sound: There is a function in the code to insert a sound in the frame, but does not work:

`Passing through key word arguments {'auth': <requests.auth.HTTPBasicAuth object at 0x7f4bf00db4e0>, 'json': {'priority': 'warning', 'model': {'frames': [{'icon': 'a7956', 'text': 'XXXXX ist unterwegs.'}], 'cycles': 1}, 'lifetime': 20000}, 'verify': False}.`

frittes commented:
@MartinHjelmare I looked at https://github.com/keans/lmnotify/blob/master/lmnotify/models.py and I think the sound has not to be added to the frame like in the ha lametric code, it has to be added to the model. Don't know, if I am right...

MartinHjelmare commented:
@frittes Thanks for good testing and feedback!

I'll have a look at the sound issue. Maybe I can do something about that, but it's for another PR. This one is ready for merging.

MartinHjelmare commented:
@frittes can you open an issue about the sound attachment not working, so we can track that, and copy your findings above into the issue? Thanks!

frittes commented:
@MartinHjelmare Thank you for making the component usable!

Yes, I will do that.

frittes commented:
Done.

https://github.com/home-assistant/home-assistant/issues/10561
